release-notes/swan-lake-GA-release-note.md

```diff
@@ -27,10 +27,238 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+   Error error(message1, code = code1, identifier = identifier1) = err; // error for 'code = code1'
+ 
+   // Similarly, attempting to bind an undefined (non-required) field `fatal` will
+   // also result in a compilation error.
+   var error(message, fatal = fatal) = err; // error for 'fatal = fatal'
+}
+```
+- Ballerina allows shadowing only variables that belong to the module-scope. A bug in the compiler that resulted in some illegal variable shadowing scenarios not being detected has been fixed.
+
+```ballerina
+function fn(int y, int z) returns int {
+   function (int, int, int) returns int f = (x, y, z) => x + y + z; // error: redeclared symbols 'y' and 'z'
+ 
+   int x = 34; // This is not in an overlapping scope, so does not result in an error.
+   return f(12, 32, 33);
+}
+ 
+function createService(int age) returns
+       service object { public function getAgeInFiveYears(int age) returns int; } {
+   return service object {
+           public function getAgeInFiveYears(int age) returns int { // error: redeclared symbol 'age'
+               return 5 + age;
+           }
+       };
+ 
+}
+```
+
+- A bug that caused classes with all `final` fields of immutable types to be considered a `readonly class` (i,e., a subtype of `readonly`) has been fixed.
+
+Such a class can no longer be used in a context that expects a subtype of `readonly`.
+
+```ballerina
+class Person {
+   final string name;
+   final string[] & readonly address;
+ 
+   function init(string name, string[] address) {
+       self.name = name;
+       self.address = address.cloneReadOnly();
+   }
+ 
+   function getAddress() returns string => string:'join(", ", ...self.address);
+}
+ 
+public function main() {
+   Person person = new ("May", ["Palm Grove", "Colombo 3"]);
+  
+   // Previously allowed, results in an error now.
+   readonly readOnlyValue = person;
+}
+```
+
+- A bug that resulted in invalid table lookups due to not distinguishing between `int`, `float`, and `decimal` zero has been fixed.
+
+```ballerina
+import ballerina/io;
+ 
+type Employee record {|
+   readonly int|float|decimal age;
+   int code;
+|};
+ 
+public function main() {
+   table<Employee> key(age) empTable = table [
+       {age: 0, code: 100} // int `0` is the key
+   ];
+ 
+   Employee? v1 = empTable[0f]; // Lookup with float `0` as the key.
+   io:println(v1 is Employee); // Previously `true`, now `false`.
+ 
+   Employee? v2 = empTable[0d]; // Lookup with decimal `0` as the key.
+   io:println(v2 is Employee); // Previously `true`, now `false`.
+ 
+   Employee? v3 = empTable[0]; // Lookup with int `0` as the key.
+   io:println(v3 is Employee); // Only this will continue to be `true`.
+}
+```
+
 To view bug fixes, see the [GitHub milestone for Swan Lake 2201.0.0](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+2201.0.0%22+label%3AType%2FBug+label%3ATeam%2FCompilerFE).
```

Comment:
Nothing related to this PR, but we dont have a milestone called 2201.0.0 right? should we update milestone to GA?

Reply:
Yes, we need to check this across all areas. 

@pubudu91, will we be changing the milestone names?

Reply:
Can change it I guess. Which would be preferable? `2201.0.0` or `2201.0.0 (Swan Lake)`? 

Also, I think we need to rephrase the above as follows: 
"To view bug fixes, see the [GitHub milestone for 2201.0.0 (Swan Lake)]"

Reply:
I prefer `2201.0.0 (Swan Lake)`. I guess we'll have to discuss this separately. 

@praneesha, please note. This will have to be updated across the release note once finalized.

Reply:
Yes, +1 for 2201.0.0 (Swan Lake) as we already have that on the [top banner in Bio](http://dev.ballerina.io/) as well.